### PR TITLE
refactor: MenuItem uses makeResetStyles

### PR DESCRIPTION
**Pull request by ling1726**
See title

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/Avatar-mount_base.html">646</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/Avatar-mount.html">629</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/Button-mount_base.html">308</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/Button-mount.html">322</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/Field-mount_base.html">1128</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/Field-mount.html">1118</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/FluentProvider-mount_base.html">682</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/FluentProvider-mount.html">692</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/FluentProviderWithTheme-mount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/FluentProviderWithTheme-mount.html">82</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">63</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">67</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">76</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/InfoButton-mount_base.html">11</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/InfoButton-mount.html">15</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/MakeStyles-mount_base.html">867</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/MakeStyles-mount.html">871</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/Persona-mount_base.html">1734</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/Persona-mount.html">1743</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/SpinButton-mount_base.html">1370</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29449/perf-test-react-components/SpinButton-mount.html">1354</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                     Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `206.455 kB`<br />`59.319 kB` |  `206.341 kB`<br />`58.82 kB` | `-114 B` <br />`-499 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          | `138.901 kB`<br />`42.923 kB` | `139.088 kB`<br />`42.549 kB` |  `187 B` <br />`-374 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             | `141.587 kB`<br />`43.437 kB` | `141.774 kB`<br />`43.058 kB` |  `187 B` <br />`-379 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                |         Size (minified/GZIP) |
| ---------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `68.748 kB`<br />`19.742 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>               | `40.866 kB`<br />`13.546 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                            |    `6.503 kB`<br />`2.22 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/93d4ee6dd801b672ade6ce41345cfbd9010af01a'>93d4ee6dd801b672ade6ce41345cfbd9010af01a</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-menu/src/components/MenuItem/useMenuItemStyles.styles.ts [hunk 1]
```diff
@@ -16,81 +16,101 @@ export const menuItemClassNames: SlotClassNames<MenuItemSlots> = {
...
+  paddingTop: tokens.spacingVerticalSNudge,
+  boxSizing: 'border-box',
+  maxWidth: '290px',
+  minHeight: '32px',
+  flexShrink: 0,
+  display: 'flex',
+  alignItems: 'start',
+  fontSize: tokens.fontSizeBase300,
+  cursor: 'pointer',
+  ...shorthands.gap('4px'),
```
Comment (layershifter) [inline packages/react-components/react-menu/src/components/MenuItem/useMenuItemStyles.styles.ts:35]:
```suggestion
  gap: '4px',
```

File: packages/react-components/react-menu/src/components/MenuItem/useMenuItemStyles.styles.ts [hunk 2]
```diff
@@ -16,81 +16,101 @@ export const menuItemClassNames: SlotClassNames<MenuItemSlots> = {
...
-    maxWidth: '290px',
-    minHeight: '32px',
-    flexShrink: 0,
-    display: 'flex',
-    alignItems: 'start',
-    fontSize: tokens.fontSizeBase300,
-    cursor: 'pointer',
-    ...shorthands.gap('4px'),
+const useRootBaseStyles = makeResetStyles({
+  ...shorthands.borderRadius(tokens.borderRadiusMedium),
```
Comment (layershifter) [inline packages/react-components/react-menu/src/components/MenuItem/useMenuItemStyles.styles.ts:20]:
```suggestion
  borderRadius: tokens.borderRadiusMedium,
```

**Review by layershifter — APPROVED** [review]

File: packages/react-components/react-menu/src/components/MenuItem/useMenuItemStyles.styles.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { mergeClasses, makeStyles, shorthands } from '@griffel/react';
+import { mergeClasses, makeStyles, shorthands, makeResetStyles } from '@griffel/react';
```
Comment (ling1726) [inline packages/react-components/react-menu/src/components/MenuItem/useMenuItemStyles.styles.ts:1]:
```suggestion
import { mergeClasses, makeStyles, makeResetStyles } from '@griffel/react';
```

**Pull request merged**